woocommerce/woocommerce
# WCA Test Helper: Product Editor Dev Tools

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the Product Editor Developer Tools to the WCA Tester Helper (part of the WooCommerce Beta Tester plugin).

**_Note: documentation on the product editor developer tools will be added in a followup PR._**

When the WooCommerce Beta Tester plugin is activated, a "Show developer tools" menu item appears in the product editor's more menu"

<img width="1073" alt="Screenshot 2023-11-06 at 16 37 42" src="https://github.com/woocommerce/woocommerce/assets/2098816/c09e1ed5-362e-4856-827f-382328fcfbe7">

The developer tools are a panel at the bottom of the editor:

<img width="1073" alt="Screenshot 2023-11-06 at 16 43 44" src="https://github.com/woocommerce/woocommerce/assets/2098816/e48860fe-5fa8-4fec-a9a7-9929bfbf7b31">

There are three tabs in the developer tools.

The _Block Inspector_ tab, which shows details for the focused block:

<img width="1073" alt="Screenshot 2023-11-06 at 16 38 38" src="https://github.com/woocommerce/woocommerce/assets/2098816/c9472c62-7481-4b27-becb-13054f6a901b">

The _Product_ tab, which shows details for the edited product, as well as the ability to test out expressions:

<img width="1073" alt="Screenshot 2023-11-06 at 16 39 15" src="https://github.com/woocommerce/woocommerce/assets/2098816/f45578e9-034e-4e79-9d3d-84a34fc3fe86">

The _Help_ tab, which shows links related to product editor development:

<img width="1073" alt="Screenshot 2023-11-06 at 16 39 20" src="https://github.com/woocommerce/woocommerce/assets/2098816/0e149a7d-5e8f-42a8-9895-e0a137707556">

Closes #40920.



### How to test the changes in this Pull Request:



1. Enable the **Product Editor** under **WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**.
2. Go to **Products** > **Add New**
3. Click on the **Show developer tools** menu item in the more menu (the kebab/ellipsis menu)
4. Verify the developer tools appear
5. Click on the close button (**X**) and/or the **Hide developer tools** menu item and verify the developer tools disappear.
6. Show the developer tools again.
7. Verify the **Block Inspector** tab works as expected. Focus on different blocks and verify the correct block info show.
8. Verify the **Product** tab works as expected. Make changes in the editor and verify the product data updates automatically.
9. Verify expression in the **Product** tab work as expected. Add an expression, such as `editedProduct.name` and verify that it updates automatically when the product is updated in the editor.
10. Verify the links in the **Help** tab go to the correct web pages, in a separate tab.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
